sale_stock_picking_validation_blocking/views/sale_order.xml

```diff
@@ -2,32 +2,35 @@
 <!-- Copyright 2020 ACSONE SA/NV
      License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
 <odoo>
+
     <record model="ir.ui.view" id="sale_order_form_view">
         <field
             name="name"
         >sale.order.form (in sale_stock_picking_validation_blocking)</field>
         <field name="model">sale.order</field>
         <field name="inherit_id" ref="sale_stock.view_order_form_inherit_sale_stock" />
         <field name="arch" type="xml">
+            <field name="warehouse_id" position="before">
+                <field name="picking_validation_blocked" />
+                <field name="hide_button_picking_validation_blocked" invisible="True" />
+            </field>
             <button name="action_draft" position="after">
                 <button
                     name="action_block_picking_validation"
                     type="object"
                     string="Block Delivery Validation"
                     groups="sales_team.group_sale_manager"
-                    attrs="{'invisible': ['|', '|', ('state', 'not in', ('sale', 'done')), ('delivery_count', '=', 0), ('picking_validation_blocked', '=', True)]}"
+                    invisible="picking_validation_blocked or hide_button_picking_validation_blocked"
```

Comment:
For better readability here, maybe use `picking_validation_blocked` in the compute method of `hide_button_picking_validation_blocked`?

Reply:
It's what I went for at the start, but there is also:
- the other button that does the opposite, and
- conditions where you want to hide both block and unblock 
- you have to reason with invisible=... instead of display=...
in that case it gets confusing since both are using the same field and you can't have a simple logic to cover all cases. I think this approach is just simple to reason with